psalm.xml

```diff
@@ -1,5 +1,7 @@
 <?xml version="1.0"?>
 <psalm
+    errorLevel="2"
+    phpVersion="8.0"
```

Comment:
I presume this also tells psalm that it should believe to be in PHP 8.0. This - as for now - is not correct, as `phive` still should run under PHP 7.3+.

Reply:
This configuration, just like PHPStan, makes the behaviour of Psalm consistent.
We don't want to have errors from Psalm depending on the PHP version we use.

Just to be clear, it doesn't mean we are dropping support for PHP 7.3.

Actually the value of this configuration is not that important, and if you wish I can replace `phpVersion="8.0"` by `phpVersion="7.3"`.
I prefered "8.0" because I though it would require less changes in the future.

Reply:
Not having looked in the documentation of either tool for the exact behavior of these options, I'm not sure I agree with your explanation - or understand it ;)

I understand the desire to have a consistent feedback. But if the suggested changes break the code for other versions of PHP this is not an option we can use. If we tell psalm it should look at things as if they'd be running in PHP 8, the changes suggested will not necessarily work with PHP 7.3 and PHP 7.4. The same would probably hold true the other way around.

I second the notion to be as much PHP 8+ optimized as possible - while still being 7.3+ compliant.

Reply:
> But if the suggested changes break the code for other versions of PHP this is not an option we can use.

Maybe I should have be more clear but the changes I've made are fully compatible with PHP 7.3.
You can see that the smoke tests and the unit tests are still passing from PHP 7.3 up to 8.1.

Anyway I've taken a different approach and now the `Curl` class doesn't have a resource as a property anymore.
That way it's no more necessary to specify a PHP version in Psalm's configuration (same for PHPStan).